analyzers/tests/SonarAnalyzer.TestFramework/Common/CombinatorialDataAttribute.cs

```diff
@@ -35,7 +35,7 @@ public DataValuesAttribute(params object[] values)
 }
 
 [AttributeUsage(AttributeTargets.Method)]
-public sealed class CombinatorialDataAttribute : Attribute, ITestDataSource
+public sealed class CombinatorialDataAttribute : TestMethodAttribute, ITestDataSource
```

Comment:
Good change, now we could add a styling rule that checks if you use `TestMehod` and `CombinatorialDataAttribute`.
There is only one case for it right now (test cases for S6932); I would include the fix in this PR.